engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -649,23 +644,17 @@ private void renderChunk(RenderableChunk chunk, ChunkMesh.RenderPhase phase, Cam
                         statRenderedTriangles += 12;
                     }
 
-                    if (shader != null) {
-                        shader.enable();
-                    }
-
                     chunk.getMesh()[i].render(phase);
                     statRenderedTriangles += chunk.getMesh()[i].triangleCount();
                 }
             }
 
+            // TODO: review - moving the deactivateFeature commands to the analog codeblock above doesn't work. Why?
             if (mode == ChunkRenderMode.DEFAULT || mode == ChunkRenderMode.REFLECTION) {
-                // eclipse is paranoid about this - it thinks that shader could be null here
-                if (shader != null) {
-                    if (phase == ChunkMesh.RenderPhase.REFRACTIVE) {
-                        shader.deactivateFeature(ShaderProgramFeature.FEATURE_REFRACTIVE_PASS);
-                    } else if (phase == ChunkMesh.RenderPhase.ALPHA_REJECT) {
-                        shader.deactivateFeature(ShaderProgramFeature.FEATURE_ALPHA_REJECT);
-                    }
+                if (phase == ChunkMesh.RenderPhase.REFRACTIVE) {
+                    chunkShader.deactivateFeature(ShaderProgramFeature.FEATURE_REFRACTIVE_PASS);
+                } else if (phase == ChunkMesh.RenderPhase.ALPHA_REJECT) {
+                    chunkShader.deactivateFeature(ShaderProgramFeature.FEATURE_ALPHA_REJECT);
```

Comment:
Just looking at the logic of these blocks, theoretically these two deactiveFeature() lines could be paired (swapped) with the analog activateFeature() lines earlier in the method. In pratice this doesn't work straight out of the box for the work done behind the scenes by the classes involved with Shaders.